src/libvirt-dbus.js

```diff
@@ -599,13 +602,31 @@ const LIBVIRT_DBUS_PROVIDER = {
         id: objPath,
         connectionName,
     }) {
+        let deviceXmlObject;
         return dispatch => {
             call(connectionName, objPath, 'org.libvirt.NodeDevice', 'GetXMLDesc', [0], { timeout, type: 'u' })
                     .then(deviceXml => {
-                        const deviceXmlObject = parseNodeDeviceDumpxml(deviceXml);
+                        deviceXmlObject = parseNodeDeviceDumpxml(deviceXml);
                         deviceXmlObject.connectionName = connectionName;
 
-                        dispatch(updateOrAddNodeDevice(deviceXmlObject));
+                        if (deviceXmlObject.path && ["pci", "usb_device"].includes(deviceXmlObject.capability.type)) {
+                            return cockpit.spawn(["udevadm", "info", "--path", deviceXmlObject.path], { err: "message" })
+                                    .then(output => {
+                                        const nodeDev = parseUdevDB(output);
+                                        if (nodeDev.SUBSYSTEM === "pci" && nodeDev) {
```

Comment:
if nodeDev does not exist the first will crash. Same for the else part.

Reply:
parseUdevDB always returns an empty object if it cannot find anything, `nodeDev.SUBSYSTEM` would not lead to a crash, just to an `undefined === "pic"`